src/constants/component-specific.ts

```diff
@@ -3,6 +3,11 @@
 /** Autocomplete */
 export const NUMBER_OF_SUGGESTION_LOADING_PLACEHOLDERS = 5;
 
+/** Button */
+export const BUTTON_TYPES = ['submit', 'button'] as const;
+
+export type ButtonTypes = typeof BUTTON_TYPES[number];
```

Comment:
Might be here enum would fit better, cause it can be both: a type and a set of values:
```ts
enum ButtonType  {
    submit = 'submit',
    button = 'button',
}
```